lenskit-predict/src/main/java/org/grouplens/lenskit/predict/KnownRatingRatingPredictor.java

```diff
@@ -0,0 +1,43 @@
+package org.grouplens.lenskit.predict;
+
+import it.unimi.dsi.fastutil.longs.LongSortedSet;
+import org.grouplens.lenskit.basic.AbstractRatingPredictor;
+import org.grouplens.lenskit.data.dao.UserEventDAO;
+import org.grouplens.lenskit.data.event.Rating;
+import org.grouplens.lenskit.vectors.MutableSparseVector;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import java.util.List;
+
+/**
+ * Created by Tajinder on 1/6/2015.
+ * 'KnownRatingRatingPredictor' predicts the known ratings of each item and the user given in the predictions
+ */
+public class KnownRatingRatingPredictor extends AbstractRatingPredictor {
+    private final UserEventDAO userEvent;
+
+    @Inject
+    public KnownRatingRatingPredictor(UserEventDAO user) {
```

Comment:
Again, naming: a user event DAO is not a 'user'. I often use 'uedao' for a short parameter name in these kinds of constructors.